pkg/reconciler/ingress/ingress.go

```diff
@@ -45,9 +45,18 @@ var _ ingress.Finalizer = (*Reconciler)(nil)
 var _ ingress.ReadOnlyFinalizer = (*Reconciler)(nil)
 
 func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
+	ing.SetDefaults(ctx)
 	before := ing.DeepCopy()
 
-	r.ObserveKind(ctx, ing)
+	if err := r.updateIngress(ctx, ing); errors.Is(err, generator.ErrDomainConflict) {
+		// If we had an error due to a duplicated domain, we must mark the ingress as failed with a
+		// custom status. We don't want to return an error in this case as we want to update its status.
+		logging.FromContext(ctx).Info("Ingress rejected as its domain conflicts with another ingress")
```

Comment:
The error message is repeated 3 times in this file. Would be good to extract it to a constant.

Reply:
Done.